vm/src/obj/objlist.rs

```diff
@@ -115,8 +113,8 @@ impl PyList {
         &self,
         vm: &VirtualMachine,
     ) -> PyResult<objbyteinner::PyByteInner> {
-        let mut elements = Vec::<u8>::with_capacity(self.get_len());
-        for elem in self.elements.borrow().iter() {
+        let mut elements = Vec::<u8>::with_capacity(self.borrow_elements().len());
+        for elem in self.borrow_elements().clone().iter() {
```

Comment:
is this cloning of Vec or lock?

Reply:
The `Vec`. `RwLock` does not implement `Clone`

Reply:
it seems additional clone is added to a few places. But this is using the iterator of the vector instead of the vector itself. Is there no way to avoid this kind of clone?

Reply:
I don't think so, if there is python code running inside the loop body, then it could theoretically lock the pylist again and cause a panic or deadlock. Although that doesn't seem necessary for this one, @palaviv, none of the code inside the loop calls into the interpreter. Is there another reason you cloned the vec here?

Reply:
I removed the `clone` here. I tried to clone only where python code is called.